R/schambach_table.R

```diff
@@ -0,0 +1,157 @@
+  start_fws,
+  start_academic_year,
+  end_fws,
+  end_academic_year,
+  complete_obsv = FALSE
+) {
+  
+#   if (missing(subgroup_cols) | missing(pretty_names)) {
+#     subgroup_cols <- c('end_schoolname')
+#     pretty_names <- c('School Name')
+#   } else {
+#     subgroup_cols <- c('end_schoolname',subgroup_cols)
+#     pretty_names <- c('School Name',pretty_names)
+#   }
+  
+  #data validation and unpack
+  assertthat::assert_that(length(subgroup_cols) == length(pretty_names))
+  
+  #unpack the mapvizieR object and limit to desired students
+  roster <- mapvizieR_obj[['roster']]
+  
+  #limit to desired grade level
+  growth_df <- mapvizieR_obj[['growth_df']]
+  growth_df <- growth_df %>% 
+    dplyr::filter(
+      end_grade == grade,
+      measurementscale == measurementscale_is
+    )
+  
+  #just desired terms
+  this_growth <- growth_df %>%
+    dplyr::filter(
+      start_map_year_academic == start_academic_year,
+      start_fallwinterspring == start_fws,
+      end_map_year_academic == end_academic_year,
+      end_fallwinterspring == end_fws
+    )
+  
+  grades_present <- unique(this_growth$start_grade)
+  
+  if(length(grades_present) > 1) {
+    warning(
+      sprintf(paste0("%i distinct grade levels present in your data! NWEA ",
+                     "school growth study tables assume cohorts composed of students ",
+                     "of the *same grade level*.  quealy_subgroups will use the mean starting grade ",
+                     "level to calculate growth scores, but you are advised to check your data and  ",
+                     "attempt to use a cohort composed of students from the same grade."), 
+              length(grades_present))
+    )
+  }
+  
+  roster <- dplyr::left_join(
+    x = roster,
+    y = this_growth[ ,c('studentid','end_schoolname')],
+    by = 'studentid'
+  )
+  
+  roster %>% 
+    ensurer::ensure_that(
+      all(subgroup_cols %in% names(roster)) ~ "subgroup_cols must match column names in your mapvizieR roster."
+    )
+  
+  if (complete_obsv == TRUE) {
+    this_growth <- this_growth %>%
+      dplyr::filter(
+        complete_obsv == TRUE  
+      )
+  }
+  
+  group_summary <- function(grouped_df) {
+    
+    df <- grouped_df %>%
+      summarize(
+        end_rit = round(mean(end_testritscore, na.rm=TRUE), digits=1),
+        start_top75 = round(100 * sum(start_testpercentile >= 75) / n(), digits=1),
+        end_top75 = round(100 * sum(end_testpercentile >= 75) / n(), digits=1),
+        avg_pg = round(mean((end_testpercentile - start_testpercentile), na.rm=TRUE), digits=1),
+        p_ku = round(100 * sum(met_typical_growth, na.rm=TRUE) / n(), digits=1),
+        p_rr = round(100 * sum(met_accel_growth, na.rm=TRUE) / n(), digits=1),
+        n = n()
+      ) %>% 
+      as.data.frame
+    
+    df
+  }
+  
+  tables <- list()
+  for (i in 1:length(subgroup_cols)) {
+    subgroup <- subgroup_cols[i]
+    
+    #join roster and data
+    if (subgroup == 'end_schoolname') {
```

Comment:
Yeah, I think what happened is that end_schoolname already exists in this_growth so if I wanted to do an inner join I would have to specify 'end_schoolname' = 'end_schoolname' if I added it as a subgroup to the minimal_roster. Otherwise, the column end_schoolname doesn't exist in the joined dataframe and an error is thrown.